packages/native-stack/src/types.tsx

```diff
@@ -342,6 +342,15 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   headerBackButtonMenuEnabled?: boolean;
+  /**
+   * How the back button behaves by default (when not customized). Available on iOS>=14.
+   * The following values are currently supported (they correspond to https://developer.apple.com/documentation/uikit/uinavigationitembackbuttondisplaymode?language=objc):
+   * - "default" – shows given back button title/previous controller title, system default or just icon based on available space
+   * - "generic" – shows given system default or just icon based on available space
+   * - "minimal" – shows just an icon
+   * @platform ios
+   */
+  backButtonDisplayMode?: ScreenStackHeaderConfigProps['backButtonDisplayMode'];
```

Comment:
All header related options are prefix with `header` so this should be changed to:

```suggestion
  headerBackButtonDisplayMode?: ScreenStackHeaderConfigProps['backButtonDisplayMode'];
```

Also if this new option is necessary, we need to remove the existing `headerBackTitleVisible` option and align JS stack to have this new option as well.

Reply:
I am unsure if we can safely remove headerBackTitleVisible or replace it. It is also kept within the react-native-screens/native-stack lib.

Reply:
The options in `react-native-screens/native-stack` are not relevant as it mostly exposes the underlying options verbatim. In our case we need to match it with other navigators, think about how different options work together and how it fits with overall system. Often we don't want to expose screen's props verbatim.